src/plugins/gs-plugin-eos.c

```diff
@@ -362,3 +362,30 @@ gs_plugin_remove_shortcut (GsPlugin	*plugin,
 	gs_app_remove_quirk (app, AS_APP_QUIRK_HAS_SHORTCUT);
 	return remove_app_from_shell (plugin, app, cancellable, error);
 }
+
+static gboolean
+app_is_flatpak (GsApp *app)
+{
+	const gchar *id = gs_app_get_id (app);
+	return id && (g_str_has_prefix (id, GS_FLATPAK_USER_PREFIX ":") ||
+		      g_str_has_prefix (id, GS_FLATPAK_SYSTEM_PREFIX ":"));
+}
+
+gboolean
+gs_plugin_app_install (GsPlugin *plugin,
+		       GsApp *app,
+		       GCancellable *cancellable,
+		       GError **error)
+{
+	if (!app_is_flatpak (app))
+		return TRUE;
+
+	/* We're only interested in already installed flatpak apps so we can
+	 * add them to the desktop */
```

Comment:
Can this be called also with apps that are not installed? It's a bit unclear to me when this function is called.